submissions/daniil-bodryagin/oop-exercise/index-oop.js

```diff
@@ -0,0 +1,180 @@
+import { print } from './js/lib.js';
+
+class Creature {
+  constructor ({name, gender, legs}) {
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this._friends = [];
+  }
+
+  set friends(listOfFriends) {
+    this._friends = listOfFriends;
+  }
+
+  get friends() {
+    return this._friends.map(({name}) => name).join(', ');
+  }
+
+  toString() {
+    const {constructor: {name: species}, name, gender, legs, friends} = this;
```

Comment:
How about setting `species` in constructor? it is definitely a property of a creature that exists not only when we want to learn anything about a creature.

Reply:
I misunderstood the requirement in "Typical mistakes" section: "1.iv : There is no need to specify species at instantiation yet species are printed". Thought that I must avoid to create property `species` in general. Fixed.

Reply:
`at instantiation` is when we do `new ClassDefinition`. Whatever is within `constructor` is already the **initialization** phase.

Reply:
Now I see. Just was confused